### DISCUSSION TITLE
send all values at startup

### DISCUSSION THREAD

**Discussion by gitrimius** [shanteacontrols/OpenDeck#44] [Ideas]
Hi,
I have noticed that when you power on a board with OpenDeck fw on it (in my case STM32F411) the current state of the controls is not spit out as midi messages. This means that the midi receiver is not aware of the current state of a certain control (or all controls for that matter) before the corresponding control has been moved.. 
For example: an oscilator on a dsp board will use its default value instead of the corresponding potentiometer value at startup..

I do understand that that is to limit the amount of data, otherwise a pot would continuously send midi data and that behaviour is ok once the controller is running.

But it would be a super nice feature to have a kind of setting for "burst all messages once, x secs after booting". That would allow the dsp to receive the values and put all internal controls on the correct pot value.
Another interesting way would be to allow for a certain midi message to come in that triggers a burst of all controller values. The latter would be more interesting in my application since it would allow to trigger the burst every time the dsp changes 'patches'/program (thus entering again the unknown value state..).
Any ideas on this?
greetings!

**Comment by paradajz** [comment]
This was actually exactly how it worked [few weeks ago](https://github.com/paradajz/OpenDeck/releases/tag/v5.0.2), but it had unfortunate side-effect that the entire board was "stuck" until USB connection was opened. On USB connection all values have been sent at once, which is good, but before that the board was unusable, which was bad (for example you couldn't use DIN MIDI or switch pages on touchscreen etc.). I will look into how to implement this again without the side effects.

**Comment by paradajz** [comment]
Now part of v5.4.0
https://github.com/paradajz/OpenDeck/releases/tag/v5.4.0

**Reply by gitrimius** [reply]
tested yesterday, works great!

Any plans to have a burst of midi controller values on demand?
If not, I'll have to find a way to reset the board easily every time I start a new patch/program on my audio DSP board (Axoloti Core). I guess I can use the reset button (or external reset pin R) for that purpose and trigger it from a GPIO from the Core...

Anyhow, great that the burst feature is back in. Thanks!

**Reply by paradajz** [reply]
That would be possible, but only via custom SysEx message. How are you intending to request the burst?

**Reply by gitrimius** [reply]
That would be great.
This is the method I would use to send the sysex message (excerpt from other forum of audio DSP):

> you can generate it using custom object or a midi script.
> using the call :
> **MidiSendSysEx(device ,port ,buffer, buflen)**;
> note: its just really a 'raw send' , ie. you need to put the enter sysex in the buffer

**EDIT**: I am reading myself in into the sysex info on the other forum and I am not fully sure it will work via DIN, usb is supported, but alas not an option for me...
I'll report back when I understand the issue better...

**Comment by paradajz** [comment]
Maybe in some future release.

-------- Original Message --------

**Reply by HBLHBL** [reply]
hi,
for my project, it would be very nice if I could send all current potentiometers values with the press of a button.
Would this be possible to implement?

**Reply by paradajz** [reply]
Yes but it's not planned at the moment.

**Reply by paradajz** [reply]
This is actually possible now if you configure a button to change the preset to the same one as the current one.

**Comment by gitrimius** [comment]
Hi,
just a short question. Suppose I recall a preset (the same that is actually already loaded) using the Program Change method. Are the values of the pots sent when changing presets? This could be an easy method the burst all values on demand...

**Reply by paradajz** [reply]
no

**Reply by gitrimius** [reply]
thanks, short question, short answer. ;-)

**Reply by paradajz** [reply]
This is now implemented:
https://github.com/paradajz/OpenDeck/commit/d56bae6216aaec0357c2489bc44b6ff0e8a29b04

It's not released yet but will be some time soon.

**Reply by gitrimius** [reply]
Hi,

That is great news.
Thanks a lot!

Will it also do a burst of the analog inputs when the same preset is called via Program Change?

**Reply by paradajz** [reply]
Yes, although this is not really the intended behaviour

**Reply by paradajz** [reply]
I will leave it like that, though, since you're finding it useful.

**Comment by paradajz** [comment]
Part of new firmware now:
https://github.com/paradajz/OpenDeck/releases/tag/v5.5.0

**Reply by gitrimius** [reply]
Thanks for implementing.
I have just tested it thoroughly. I works great but I think I found an anomaly.
All values for analog inputs and buttons/switches are now updated on a Preset Change. But there is 1 case that is still missing, I think:
An analog input that is configured to 'Button' is probably not part of the burst. At least when I change Presets the led that is linked (through Internal Note message) is not updated. When the switch is switched on and off after a Preset change the led is updated correctly
I am building from master v5.5.0.

**Reply by paradajz** [reply]
Yes, it's not part of it. Not by design, though. I'll solve this in the next release which will also bring brightness control to LEDs.

**Reply by paradajz** [reply]
The fix is now on master.

**Reply by paradajz** [reply]
New firmware is now out: https://github.com/paradajz/OpenDeck/releases/tag/v5.5.1

**Reply by gitrimius** [reply]
Hi, 
I have a question concerning the burst of the analog ins configured as Button.
Since I built from master (for STM32) v 5.5.3, this button burst is not happening anymore.  (It ìs working for normal analogs ins & Buttons)
Is there a possibility it is not activated in this version? 
(Note: I did build of the (shortly available) v5.5.4. But I saw that 5.5.4 was replaced by 5.5.3, I downgraded. Don't know if it could have something to do with that...)

**Reply by paradajz** [reply]
Which burst? Startup or when changing presets?

**Reply by gitrimius** [reply]
I have 8 switches connected. 7 on digital pins, 1 on analog.

when changing **presets** I get:
_NoteOn  ch16 n  8 v127
NoteOn  ch16 n  7 v127
NoteOn  ch16 n  6 v127
NoteOn  ch16 n  5 v127
NoteOn  ch16 n  4 v127
NoteOn  ch16 n  3 v127
NoteOn  ch16 n  2 v  0_

At **startup** I get the following:

_NoteOn  ch16 n  8 v  0
NoteOn  ch16 n  7 v  0
NoteOn  ch16 n  6 v  0
NoteOn  ch16 n  5 v  0
NoteOn  ch16 n  4 v  0
NoteOn  ch16 n  3 v  0
NoteOn  ch16 n  2 v  0
NoteOn  ch16 n  8 v127
NoteOn  ch16 n  7 v127
NoteOn  ch16 n  6 v127
NoteOn  ch16 n  5 v127
NoteOn  ch16 n  4 v127
NoteOn  ch16 n  3 v127
NoteOn  ch16 n  1 v127_

**Comment by gh0stless** [comment]
Is it possible to make this burst on a sysex request?

**Reply by gh0stless** [reply]
Maybe you should also give this sysex command a value for a delay as a parameter.
I can also confirm that the way it is now, it seems to be delivering incorrect values.

**Reply by paradajz** [reply]
What do you mean with incorrect values?

**Reply by gh0stless** [reply]
Hello and thank you,

sorry for the delay, but I had to find time to test it properly again.

I tested it with MIDI-OX. Apparently, when connecting via USB, all controllers come up first, but with incorrect values ​​(e.g. pan), then the notes, and then controllers come up that I don't use at all.
